Sources/TensorFlow/Operators/Array.swift

```diff
@@ -0,0 +1,103 @@
+// Copyright 2019 The TensorFlow Authors. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+
+public extension Tensor where Scalar: TensorFlowNumeric {
+    /// Returns the diagonal part of the tensor.
+    ///
+    /// For example:
+    ///
+    /// ```
+    /// // 't' is [[1, 0, 0, 0]
+    /// //         [0, 2, 0, 0]
+    /// //         [0, 0, 3, 0]
+    /// //         [0, 0, 0, 4]]
+    /// t.diagonalPart()
+    /// // [1, 2, 3, 4]
+    ///
```

Comment:
minor: missing end backticks:
```suggestion
    /// ```
```